CXP-934 Added scope documentation (#536)
akeneo/pim-api-docs: content/apps/oauth2-scopes.md

# OAuth scopes

> During App authorization, App must specify the minimal set of scopes that is required for the App to function.

Reviewer: ```suggestion
Part of the app authorization process requires specifying which parts of a PIM's data the app needs access to.
```

> During App authorization, App must specify the minimal set of scopes that is required for the App to function.

Reviewer: ```suggestion
Part of the app authorization process requires specifying which parts of a PIM's data the app needs access to.
```

Reviewer: needs != requires

In the OAuth spec, you request scopes you need and the Authorization Server decides which ones will be given to you.
You can totally imagine that each requested scope can be checked/unchecked by the user granting you access.
That's why you have to check, in the Access Token Response which scopes were granted to your App.

An App cannot *requires* scopes to work.

Reviewer: @willy-ahva https://shopify.dev/api/usage/access-scopes

> Scopes limits App access to the **PIM REST API**. Once App is authorized, it can only interact with API when in compliance with scopes requested.

Reviewer: I think this is obvious and we should remove this.

::: warning
Scopes cannot be changed once activation process is over.

> If your App needs to update the list of required scopes, it must redo the process of activation.

Reviewer: This is confusing and should be clarified because an App cannot redo by itself the process of activation.

::::

## How to use scopes

The list of scopes is specified by the `scope` query parameter on PIM authorize endpoint.
Scopes list should be a string separted with spaces.

#### example of scopes use (in this case the space is URL-encoded as `%20`)

```
https://my-pim.cloud.akeneo.com/connect/apps/v1/authorize?
    response_type=code&
    client_id=[OAUTH_CLIENT_ID]&
    redirect_uri=[CALLBACK_URL]&
    scope=write_products%20read_catalog_structure&
    state=[STATE]
```

Any erroneous scope values are ignored

> Authorization response will return a list of scopes that user gave consent

Reviewer: I think you should remove this part.

Also,
> Authorization response will return a list of scopes that user gave consent

This is a good reminder, but I think this information should go to the OAuth documentation.

## Supported scopes

> When an app specifies a scope from the list below, upon activation the app will recieve a token that can only act within boundairies of the granted access

Reviewer 2: ```suggestion
When an app specifies a scope from the list below, upon activation the app will receive a token that can only act within the boundaries of the granted access
```

| Scope | Grants access to |
|-------|------------|

> | (no scope) | Any endpoint not bounded by any permissions |

Reviewer: ```suggestion
```

I would remove this line, it will create more questions than necessary. Maybe I'm wrong.

> | read_products | Read products |
> | write_products | Read and write products |
> | delete_products | Remove products |

Reviewer: ```suggestion
| read_products | Read products and product models |
| write_products | Read and write products and product models |
| delete_products | Remove products and product models |
```

> | read_catalog_structure | Read attributes, attribute groups, families and family variants|

Reviewer: ```suggestion
| read_catalog_structure | Read attributes, attribute groups, families and family variants |
```

| write_catalog_structure | Read and write attributes, attribute groups, families and family variants |
| read_attribute_options | Read attribute options |

> | write_attribute_options | Read and write attribute options  |

Reviewer: ```suggestion
| write_attribute_options | Read and write attribute options |
```

> | read_categories | Read categories  |

Reviewer: ```suggestion
| read_categories | Read categories |
```
